src/main/webapp/script.js

```diff
@@ -213,20 +216,284 @@ function getGraphDisplay(graphNodes, graphEdges) {
+  // Initialize empty collections
+  let deletedNodes = cy.collection();
+  let deletedEdges = cy.collection();
+  let addedNodes = cy.collection();
+  let addedEdges = cy.collection();
+  let modifiedNodes = cy.collection();
+
+  // If the mutation list is null/undefined
+  if (!mutList) {
+    return [deletedNodes, deletedEdges, addedNodes, addedEdges, modifiedNodes];
+  }
+  // Apply each mutation
+  mutList.forEach(mutation => {
+    const type = mutation["type_"] || -1;
+    const startNode = mutation["startNode_"];
+    const endNode = mutation["endNode_"];
+    let modifiedObj = cy.collection();
+
+    if (!type || !startNode) {
+      return modifiedObj;
+    }
+
+    switch (type) {
+      case 1:
+        // add node
+        if (cy.getElementById(startNode).length !== 0) {
+          modifiedObj = cy.getElementById(startNode);
+          // color this node green
+          modifiedObj.style('background-color', 'green');
+          addedNodes = addedNodes.union(modifiedObj);
+        }
+        break;
+      case 2:
+        // add edge
+        if (endNode && cy.getElementById(startNode).length !== 0 && cy.getElementById(endNode).length !== 0) {
+          modifiedObj = cy.getElementById(`edge${startNode}${endNode}`);
+          // color this edge green
+          modifiedObj.style('line-color', 'green');
+          modifiedObj.style('target-arrow-color', 'green');
+          addedEdges = addedEdges.union(modifiedObj);
+        }
+        break;
+      case 3:
+        // delete node
+        // add a phantom node (if it doesn't already exist) and color it red
+        if (cy.getElementById(startNode).length === 0) {
+          cy.add({
+            group: "nodes",
+            data: { id: startNode }
+          });
+        }
+        modifiedObj = cy.getElementById(startNode);
+        modifiedObj.style('background-color', 'red');
+        modifiedObj.style('opacity', 0.25);
+        deletedNodes = deletedNodes.union(modifiedObj);
+        break;
+      case 4:
+        // delete edge
+        if (!endNode) {
+          break;
+        }
+        // if corresponding nodes don't exist, add them
+        if (cy.getElementById(startNode).length === 0) {
+          cy.add({
+            group: "nodes",
+            data: { id: startNode }
+          });
+        }
+        if (cy.getElementById(endNode).length === 0) {
+          cy.add({
+            group: "nodes",
+            data: { id: endNode }
+          });
+        }
+        // Add a phantom edge and color it red
+        cy.add({
+          group: "edges",
+          data: {
+            id: `edge${startNode}${endNode}`,
+            target: endNode,
+            source: startNode
+          }
+        });
+        modifiedObj = cy.getElementById(`edge${startNode}${endNode}`);
+        modifiedObj.style('line-color', 'red');
+        modifiedObj.style('target-arrow-color', 'red');
+        modifiedObj.style('opacity', 0.25);
```

Comment:
you could also do the constants thing for opacity, so if you want to change it from .25 to .3 for ex, you would only have to change it in the constant rather than everywhere it says .25